scripts/check_espn_api.py

```
@@ -0,0 +1,362 @@
+
+
+def probe(url, headers, timeout):
+    """Fetch url and return a result dict. Never raises."""
+    # The URLs come from the table above, but pin the scheme rather than trusting
+    # them: urlopen would honour file:// or a custom scheme if an entry ever
+    # arrived from somewhere less trustworthy. Reported as an ordinary failure so
+    # the promise above holds and one bad row cannot abort the whole run.
+    if not url.startswith("https://"):
+        return {"ok": False, "status": None, "error": f"refusing non-HTTPS URL: {url!r}"}
+
+    request = urllib.request.Request(url)
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)